### DISCUSSION TITLE
Animations in JointJS

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2409] [Q&A] [answered]
### Introduction

Hi guys, is there any documentation for how to apply ,render and remove animations on JointJS elements?,if there is one can anyone send me the link for that documentation

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
There are several ways to animate elements:

1. **Web Animations API**
   - Use a native API inside a custom views.
   _docs_: [MDN Web Animations API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API) and [custom views I.](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.CellView.custom) abd [custom views II.](https://resources.jointjs.com/tutorial/content-driven-element)
   _examples_: [scada](https://www.jointjs.com/demos/scada), [yamazumi](https://www.jointjs.com/demos/yamazumi-3d), [electric generator](https://www.jointjs.com/demos/electric-generator)

2. **CSS Transitions**
   - Add CSS to animate DOM nodes.
  _docs_:  [MDN CSS Transitions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_transitions)
  _examples_: [process flow diagram](https://www.jointjs.com/demos/animated-process-flow-diagram)

3. **Cell Model Transitions**
   - Change of model attributes over time.
  _docs_: [Cell.prototype.transition()](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Element.prototype.transition)
  _examples_:  [puzzle](https://www.jointjs.com/demos/puzzle),  [tokens](https://www.jointjs.com/demos/tokens)

4. **Send token along a path**
   - Animate a DOM nodes along an SVG Path.
  _docs_: [LinkView.prototype.sendToken()](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.LinkView.prototype.sendToken) and low-level [V.prototype.animateAlongPath()](https://resources.jointjs.com/docs/jointjs/v3.7/vectorizer.html#V.prototype.animateAlongPath).
  _examples_: [petri nets](https://www.jointjs.com/demos/petri-nets)

**Reply by VADDIJASWANT** [reply]
thanks a lot
